File: Domain/Purchase/src/main/java/com/ecommerce/Domain/Domain/Purchase.java
package com.ecommerce.Domain.Domain;

import lombok.*;

import java.util.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Purchase {
    private Long id;

    private PaymentMethod payment;

    private Long customerId;

    private String state;

    private Date createdAt;

    private Date updatedAt;

    private List<PurchaseProduct> purchaseProducts;

}

File: Client/src/interfaces/purchase.ts
import { CustomerInterface } from "./customer";
import { PurchasesProductsInterface } from "./purchasesProducts";

export interface PurchaseInterface {
  id?: number;
  state: string;
  payment: string;
  customerId: CustomerInterface | number;
  purchaseProducts?: PurchasesProductsInterface[];
  totalPurchase?:number;
  createdAt?:string;
}
